File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/UserSettings.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*

class UserSettings {
    suspend fun getUserSettings(sdk: BreezSdk) {
        // ANCHOR: get-user-settings
        try {
            val userSettings = sdk.getUserSettings()
            println("User settings: $userSettings")
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: get-user-settings
    }

    suspend fun updateUserSettings(sdk: BreezSdk) {
        // ANCHOR: update-user-settings
        // Fields left as null are not changed. Settings that take an enum value
        // accept Set to assign one, or Unset to clear it.
        try {
            sdk.updateUserSettings(UpdateUserSettingsRequest(
                sparkPrivateModeEnabled = true,
                stableBalanceActiveLabel = null,
                sparkMasterIdentityPublicKey = SparkMasterIdentityPublicKey.Set(
                    publicKey = "<hex encoded public key>"
                )
            ))
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: update-user-settings
    }

    suspend fun activateStableBalance(sdk: BreezSdk) {
        // ANCHOR: activate-stable-balance
        try {
            sdk.updateUserSettings(UpdateUserSettingsRequest(
                sparkPrivateModeEnabled = null,
                stableBalanceActiveLabel = StableBalanceActiveLabel.Set(label = "USDB")
            ))
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: activate-stable-balance
    }

    suspend fun deactivateStableBalance(sdk: BreezSdk) {
        // ANCHOR: deactivate-stable-balance
        try {
            sdk.updateUserSettings(UpdateUserSettingsRequest(
                sparkPrivateModeEnabled = null,
                stableBalanceActiveLabel = StableBalanceActiveLabel.Unset
            ))
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: deactivate-stable-balance
    }
}

File: docs/breez-sdk/snippets/swift/BreezSdkSnippets/Sources/UserSettings.swift
import BreezSdkSpark

func getUserSettings(sdk: BreezSdk) async throws {
    // ANCHOR: get-user-settings
    let userSettings = try await sdk.getUserSettings()
    print("User settings: \(userSettings)")
    // ANCHOR_END: get-user-settings
}

func updateUserSettings(sdk: BreezSdk) async throws {
    // ANCHOR: update-user-settings
    // Fields left as nil are not changed. Settings that take an enum value
    // accept set to assign one, or unset to clear it.
    try await sdk.updateUserSettings(
        request: UpdateUserSettingsRequest(
            sparkPrivateModeEnabled: true,
            stableBalanceActiveLabel: nil,
            sparkMasterIdentityPublicKey: .set(publicKey: "<hex encoded public key>")
        ))
    // ANCHOR_END: update-user-settings
}

func activateStableBalance(sdk: BreezSdk) async throws {
    // ANCHOR: activate-stable-balance
    try await sdk.updateUserSettings(
        request: UpdateUserSettingsRequest(
            sparkPrivateModeEnabled: nil,
            stableBalanceActiveLabel: .set(label: "USDB")
        ))
    // ANCHOR_END: activate-stable-balance
}

func deactivateStableBalance(sdk: BreezSdk) async throws {
    // ANCHOR: deactivate-stable-balance
    try await sdk.updateUserSettings(
        request: UpdateUserSettingsRequest(
            sparkPrivateModeEnabled: nil,
            stableBalanceActiveLabel: .unset
        ))
    // ANCHOR_END: deactivate-stable-balance
}
